run.go

```diff
@@ -326,15 +332,63 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := make([]string, 0, len(s))
+	seen := make(map[string]string)
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// checkForDerivitives creates a slice of unique model variables that are
+// required to calculate the requested outputVariables.
+func checkForDerivitives(m map[string]string, f map[string]govaluate.ExpressionFunction) (map[string]string, []string, error) {
+	getVariables := make([]string, 0, len(m))
+	for key, val := range m {
+		expression, err := govaluate.NewEvaluableExpressionWithFunctions(val, f)
+		if err != nil {
+			return nil, nil, fmt.Errorf("inmap: unsupported expression '%s'", val)
+		}
+		uniqueVars := removeDuplicates(expression.Vars())
+		getVariables = append(getVariables, uniqueVars...)
+		for _, uniqueVar := range uniqueVars {
+			if m[uniqueVar] != "" && m[uniqueVar] != uniqueVar {
+				m[key] = strings.Replace(m[key], uniqueVar, "("+m[uniqueVar]+")", -1)
+				return checkForDerivitives(m, f)
+			}
+		}
+	}
+	return m, removeDuplicates(getVariables), nil
+}
+
+func (d *InMAP) checkGetNames(g ...string) error {
```

Comment:
Please add a documentation comment here (even though there wasn't one before [because there should have been one here before]).